src/Domain/DomainHelpers.hpp

```diff
@@ -0,0 +1,289 @@
+ *
+ *  ----------------------------------------------------------------------
+ *
+ */
+
+#pragma once
+
+#include <algorithm>
+#include <numeric>
+
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+
+template <size_t>
+class BlockNeighbor;
+
+template <size_t>
+class Direction;
+
+template <size_t, size_t>
+class EmbeddingMap;
+
+template <size_t VolumeDim>
+struct DomainHelpers {
+  ///\returns the common corners in the global corner numbering scheme.
+  static std::vector<size_t> get_common_global_corners(
+      const std::array<size_t, two_to_the(VolumeDim)>& block_1_global_corners,
+      const std::array<size_t, two_to_the(VolumeDim)>& block_2_global_corners);
+
+  ///\returns the common corners in the local corner numbering scheme.
+  template <typename Seq>
+  static std::array<size_t, two_to_the(VolumeDim - 1)> get_common_local_corners(
+      const std::array<size_t, two_to_the(VolumeDim)>& block_global_corners,
+      const Seq& block_global_common_corners);
+
+  ///\requires that the ids be those of corners in the local numbering scheme.
+  ///\returns the Points corresponding to those corners in the Logical grid.
+  static std::array<Point<VolumeDim, Frame::Logical>, two_to_the(VolumeDim - 1)>
+  logical_coords_of_corners(
+      const std::array<size_t, two_to_the(VolumeDim - 1)>& local_ids);
+
+  /// \requires that the Points correspond to the corners of a face in the
+  /// Logical grid.
+  /// \returns the corresponding Direction of the outward normal to the face
+  /// defined by those corners.
+  static Direction<VolumeDim> get_direction_normal_to_face(
+      const std::array<Point<VolumeDim, Frame::Logical>,
+                       two_to_the(VolumeDim - 1)>& face_pts);
+
+  /// \requires that the ids correspond to the corners of a face ordered in the
+  /// local numbering scheme.
+  /// \returns the corresponding Directions of the axes along the face defined
+  /// by those corners.
+  static std::vector<Direction<VolumeDim>> get_directions_along_face(
+      const std::array<size_t, two_to_the(VolumeDim - 1)>& face_global_ids);
+
+  /// Takes a Direction in block1 and returns the corresponding Direction
+  /// in block 2.
+  static Direction<VolumeDim> mapped(
+      size_t block1_id, size_t block2_id,
+      const Direction<VolumeDim>& dir_in_block1,
+      const std::vector<std::array<size_t, two_to_the(VolumeDim)>>&
+          corners_of_all_blocks);
+
+  /// Takes the Directions in block1 and finds the corresponding
+  /// Directions in block2 according to their corner numberings.
+  static std::pair<std::array<Direction<VolumeDim>, VolumeDim>,
+                   std::array<Direction<VolumeDim>, VolumeDim>>
+  obtain_correspondence_between_blocks(
+      size_t block1_id, size_t block2_id,
+      const std::vector<std::array<size_t, two_to_the(VolumeDim)>>&
+          corners_of_all_blocks);
+
+  /// Given Directions in block1 and their corresponding Directions
```

Comment:
What's done first is that the Direction and the vector<Direction> are packaged into an array. This is done for each block, then the two arrays are made into a pair - I'll edit the comment